rfcs/52-unstable_features.md

```diff
@@ -0,0 +1,65 @@
+# Feature Name: `unstable_features`
+
+## Summary
+
+Accepting PRs for new features that may not have reach the high bar we set for Bevy, and disable their changes with a feature gate.
+
+## Motivation
+
+Some features would benefit from iterative work in Bevy, but that may be hard to do when a PRs need a high level of quality to be merged in.
+
+In this RFC, "feature" means the new feature or change under development, and "feature gate" the conditional compilation flag. "unstable" means a feature that is subject to change, not something that will break Bevy.
+
+## User-facing explanation
+
+So you want to contribute a new feature to Bevy! You have a few paths available to you, depending on the scope of the feature:
+- A small feature with a clear implementation? Go ahead and open that PR!
+- A large feature with an open design space that could use some discution about how to implement it? A RFC would be good.
+- Something in between that would benefit from iterative work? Here come the unstable feature workflow for you!
+
+You will need to open a tracking issue to explain the new feature you want to add. If it looks good, you'll be ready to open that PR with a feature gate for your changes!
+
+## Implementation strategy
```

Comment:
I think a convention for feature flags naming should be kept so that users can quickly discern which features are unstable, and we should have some log or central location to track these features.

Also unfortunately feature flag doc generation are currently bound to nightly. Not sure how to get around this before that feature gets stabilized.

Reply:
Agreed on the convention.

I think it's good that the doc won't be built for those unstable features by default. If someone wants it, the doc can be built on demand with the feature gate enabled.

Reply:
On one hand, I'm inclined to agree that we shouldn't be putting this forth front and center, but that also hides these features from those who might both benefit from them in the short term and be a good testbed for stabilization. By showing them and explicitly labeling APIs as "unstable, expect bugs", I think that should deter any user expecting stability. The same goes for nightly-bound unstable features in Rust: it garners both testers and active feedback.

Reply:
I think a GitHub project showing tracking the stabilisation issues should be good enough for discovery, what do you think?
I added that proposition to the RFC. Keeping the project up to date could be done by GitHub Actions, but I think that's out of scope of this RFC.

Reply:
> I think a GitHub project showing tracking the stabilisation issues should be good enough for discovery

For comparison, Rust has the [RFC book](https://rust-lang.github.io/rfcs/). A GitHub Project might be good enough as a start, but IMO if we want this to be a publicly recognizable format, we need more than that in the long run.

Reply:
Having something like the RFC book requires quite a bit more work...

It is needed in Rust where they need to keep track of everything back to the pre 1.0 time for people still using an old version, they have feature opened 7 years ago not yet closed, and they track a lot of them.
This is not what I want to Bevy, where I would rather have around 10 unstable features open at any given time, and for no more than a year.

I think filtering issues on the label, and presenting them in a project would be enough for some time

Reply:
> I think a GitHub project showing tracking the stabilisation issues should be good enough for discovery, what do you think?

This is hard to discover for Bevy users which are not contributors.